src/server/serve.rs

```diff
@@ -411,7 +411,7 @@ impl InnerService {
             if let Some(encoding) = req.headers().get(hyper::header::ACCEPT_ENCODING) {
                 let content_encoding = get_prior_encoding(encoding);
                 if should_compress(content_encoding) {
-                    let b = compress_stream(
+                    let body = compress_stream(
```

Comment:
Given `body` is already mutable, could we just bind value to existing `body` variable? That would shave off at least three `else` branches.

Reply:
@weihanglo I have simplified branches with mutation on  4b3069c